ui/core/src/utils/time-series-data.ts

```diff
@@ -65,28 +65,68 @@ export function getYValues(series: TimeSeries, timeScale: TimeScale): Array<numb
  * the x axis (i.e. start/end dates and a step that is divisible into all of
  * the queries' steps).
  */
-export function getCommonTimeScale(seriesData: Array<TimeSeriesData | undefined>): TimeScale | undefined {
+export function getCommonTimeScale(
+  seriesData: Array<TimeSeriesData | Pick<TimeSeries, 'values'> | undefined>
+): TimeScale | undefined {
   let timeRange: AbsoluteTimeRange | undefined = undefined;
   const steps: number[] = [];
-  for (const data of seriesData) {
-    if (data === undefined || data.timeRange === undefined || data.stepMs === undefined) continue;
-
-    // Keep track of query steps so we can calculate a common one for the graph
-    steps.push(data.stepMs);
-
-    // If we don't have an overall time range yet, just start with this one
-    if (timeRange === undefined) {
-      timeRange = data.timeRange;
-      continue;
-    }
 
-    // Otherwise, see if this query has a start or end outside of the current
-    // time range
-    if (data.timeRange.start < timeRange.start) {
-      timeRange.start = data.timeRange.start;
-    }
-    if (data.timeRange.end > timeRange.end) {
-      timeRange.end = data.timeRange.end;
+  for (const data of seriesData) {
+    if (data && 'timeRange' in data) {
+      if (data === undefined || data.timeRange === undefined || data.stepMs === undefined) continue;
+
+      // Keep track of query steps so we can calculate a common one for the graph
+      steps.push(data.stepMs);
+
+      // If we don't have an overall time range yet, just start with this one
+      if (timeRange === undefined) {
+        timeRange = data.timeRange;
+        continue;
+      }
+
+      // Otherwise, see if this query has a start or end outside of the current
+      // time range
+      if (data.timeRange.start < timeRange.start) {
+        timeRange.start = data.timeRange.start;
+      }
+      if (data.timeRange.end > timeRange.end) {
+        timeRange.end = data.timeRange.end;
+      }
+    } else if (data && 'values' in data) {
```

Comment:
added this so we could also use this function in TimeChart component to get a default time scale

Reply:
Makes sense! So in a future release once we confirm the new chart works as expected, we can remove the logic above and only keep this new if condition (and remove the `show_legacy_chart` panel option / other conditions for LineChart).